controllers/storage.go

```diff
@@ -98,13 +100,51 @@ func getHttpsTransportWithCACert(log logr.Logger, pemCerts [][]byte) (*http.Tran
 	return transport, nil
 }
 
+func configureProxyForTransport(transport *http.Transport, proxyConfig *dspav1.ProxyConfig) {
+	if proxyConfig == nil {
+		return
+	}
+	cfg := httpproxy.Config{
+		HTTPProxy:  proxyConfig.HTTPProxy,
+		HTTPSProxy: proxyConfig.HTTPSProxy,
+		NoProxy:    proxyConfig.NoProxy,
+	}
+	proxyFn := cfg.ProxyFunc()
+	transport.Proxy = func(req *http.Request) (*url.URL, error) {
+		return proxyFn(req.URL)
+	}
+}
+
+func getTransportWithProxyAndCACert(log logr.Logger, pemCerts [][]byte, proxyConfig *dspav1.ProxyConfig, secure bool) (*http.Transport, error) {
+	var transport *http.Transport
+	var err error
+
+	if secure {
+		transport, err = getHttpsTransportWithCACert(log, pemCerts)
```

Comment:
Is there a downside to always using this even if Minio is not over HTTPS? The issue is that even though Minio might not be HTTPS, the proxy could be so we should include `pemCerts` in that case.

Reply:
Agree with that, I updated to use the certs even when Minio isn’t running over HTTPS and tested, it worked without any errors.